scripts/kconfig/menuconfig.py

```diff
@@ -1937,20 +1990,44 @@ def select_prev_match():
             s, s_i, hscroll = _edit_text(c, s, s_i, hscroll,
                                          edit_box.getmaxyx()[1] - 2)
 
-# Obscure Python: We never pass a value for cached_search_nodes, and it keeps
-# pointing to the same list. This avoids a global.
-def _searched_nodes(cached_search_nodes=[]):
-    # Returns a list of menu nodes to search, sorted by symbol name
+# Obscure Python: We never pass a value for cached_nodes, and it keeps pointing
+# to the same list. This avoids a global.
+def _sorted_sc_nodes(cached_nodes=[]):
+    # Returns a sorted list of symbol and choice nodes to search. The symbol
+    # nodes appear first, sorted by name, and then the choice nodes, sorted by
+    # prompt and (secondarily) name.
 
-    if not cached_search_nodes:
-        # Sort symbols by name, then add all nodes for each symbol
+    if not cached_nodes:
+        # Add symbol nodes
         for sym in sorted(_kconf.unique_defined_syms,
                           key=lambda sym: sym.name):
-
             # += is in-place for lists
-            cached_search_nodes += sym.nodes
+            cached_nodes += sym.nodes
+
+        # Add choice nodes
+
+        choices = sorted(_kconf.choices, key=lambda choice: choice.name or "")
```

Comment:
Just thought of a minor bug re. named choices defined in multiple locations here. They get listed twice, because the code should use `unique_choices` instead of `choices`.

Working on West stuff atm, but I'll push a fix a bit later.

Reply:
Fixed